documentation/modules/advanced/pages/diferencia.adoc

```diff
@@ -0,0 +1,239 @@
+if so run:
+
+[source, bash]
+----
+./scripts/clean.sh
+----
+====
+
+[#what-is-diferencia]
+== What is Diferencia?
+
+https://lordofthejars.github.io/diferencia-docs-site[*Diferencia*] is designed to act as a proxy which intercepts calls to a service and multicast it to several instances of the same service. 
+These instances are an old and new version of the service. Then when the response is returned from each instance, Diferencia will check if both responses are _similar_, and if it is the case, then the two implementations might be considered compatible and the new version implementation is *regression-free*.
+
+image::diferencia-simple.png[Overview of Diferencia]
+
+In the previous schema, you can see that a request is multicasted to two instances of Service A, one being V1 and to other V2. 
+Both return a response and then it is compared by Diferencia. 
+Finally, notice that no response is returned (_by default_) but the result (in form of HTTP Status Code).
+
+Since Diferencia does not return a real response, but a comparison, effectively means that it is a perfect fit for *Traffic Shadowing/Mirroring* technique.
+
+[#master-candidate]
+=== Selecting Master and Candidate
+
+The first thing you need to do is selecting a *master* service (the one you know it works) and a *candidate* service (the one you want to release) and create a Kubernetes service.
+For this case, *master* is `recommendation:v1` and *candidate* is `recommendation:v2`.
+
+[source, bash,subs="+macros,+attributes"]
+----
+oc create -f link:{github-repo}/{diferencia-dir}/recommendation-master-svc.yml[diferencia/recommendation-master-svc.yml] -n tutorial
+oc create -f link:{github-repo}/{diferencia-dir}/recommendation-candidate-svc.yml[diferencia/recommendation-candidate-svc.yml] -n tutorial
+
+or
+
+kubectl create -f link:{github-repo}/{diferencia-dir}/recommendation-master-svc.yml[diferencia/recommendation-master-svc.yml] -n tutorial
+kubectl create -f link:{github-repo}/{diferencia-dir}/recommendation-candidate-svc.yml[diferencia/recommendation-candidate-svc.yml] -n tutorial
+----
+
+[#deploying-diferencia]
+=== Deploying Diferencia
+
+Next thing is to deploy Diferencia.
+
+[source, bash]
+----
+oc apply -f <(istioctl kube-inject -f link:{github-repo}/{diferencia-dir}/Deployment.yml[diferencia/Deployment.yml]) -n tutorial
+oc create -f diferencia/Service.yml -n tutorial
+
+or
+
+kubectl apply -f <(istioctl kube-inject -f link:{github-repo}/{diferencia-dir}/Deployment.yml[diferencia/Deployment.yml]) -n tutorial
+kubectl create -f diferencia/Service.yml -n tutorial
+----
+
+[#shadowing-traffic]
+=== Shadowing Traffic
+
+The last thing is to just start shadowing traffic between `recommendation:v1` and diferencia.
+
+[source, bash,subs="+macros,+attributes"]
+----
+istioctl create -f link:{github-repo}/{diferencia-dir}/virtual-service-recommendation-v1-diferencia-mirror-v2.yml[diferencia/virtual-service-recommendation-v1-diferencia-mirror-v2.yml] -n tutorial
```

Comment:
It misses the destination rule file.

Reply:
I couldn't proceed because it fails without the destination rule

Reply:
The file is there probably the problem is that I am assuming user is in a directory which he isn't there. Let me review.

Reply:
In theory it is correct. I have tried and it finds it, can you do a `pwd` and check you are in the root of `istio-tutorial?`

Reply:
The destinatio-rule is not mentioned in the documentation. Only the virtual-service.

Reply:
true, adding it.